### PR TITLE
Storage of data

### PR DESCRIPTION
**Pull request by yegonkimutai**
In this PR, my partner and I
- [ ] Create a save method that will save the data into the respectable file when they are added.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -120,4 +121,16 @@ def list_rentals
...
+    save_data_to_json('books.json', @books)
+    save_data_to_json('people.json', @person)
+    save_data_to_json('rentals.json', @rentals)
+  end
+
+  private
+
+  def save_data_to_json(filename, data)
+    File.write(filename, JSON.generate(data))
+  end
```
Comment (mckent05) [inline app.rb]: - Good job done so far, but you are yet to implement all projects requirements, Part of the project requirements say:
`On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing`. Please fix this 🙏

Comment (mckent05) [inline app.rb]: - Good job done so far, but you are yet to implement all projects requirements, Part of the project requirements say:
`On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing`. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi guys, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job writing functions to store your data 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 2]
```diff
@@ -120,4 +121,31 @@ def list_rentals
...
+    save_data_to_json('people.json', @person)
+    save_data_to_json('rentals.json', @rentals)
+  end
+
+  def load_data
+    @books = load_data_from_json('books.json', [])
+    @people = load_data_from_json('people.json', [])
+    @rentals = load_data_from_json('rentals.json', [])
+  end
+
```
Comment (TedLivist) [inline app.rb]: - [ ] You have not handled the scenario of saving books, properly. Your implementation saved the memory of the book and people, and when I start the app, the JSON files are cleared. I am unable to list the books. Kindly fix that. Your JSON files should keep track of all the data, regardless of whether the app was exited or not. Previously added data should still be present on loading the app.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @yegonkimutai and @AbbasSarwar,

✔️ Preservation of data

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -122,30 +122,72 @@ def list_rentals
...
-    save_data_to_json('rentals.json', @rentals)
+  def get_data(file_name)
+    if File.exist?("#{file_name}.json")
+      File.read("#{file_name}.json")
+    else
+      empty_json = [].to_json
+      File.write("#{file_name}.json", empty_json)
+      empty_json
+    end
   end
```
Comment (TadesseDev) [inline app.rb]: - [ ] Dear @yegonkimutai, as suggested by the previous reviewer and as stated clearly in the project requirement, On exit from your app we need to make sure the data is being saved (all the data: books, people, and rentals) into JSON files. In your case after exiting the app, no data is being saved to the JSON files, and no data is retrieved on the app start. Please make sure data are always saved to the JSON files and retrieved on the app start.

### This is my experience
image

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @yegonkimutai 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] Good job making most of the changes requested by the previous reviewer. 🙏


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi  Abbas & Yegon,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Data is stored correctly in JSON on program exit!
🎲 ReadMe is near Professional!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

**You can also consider:**
Changing the texts _author1_ and _author2_ under the Authors section of your ReadMe to your respective full names will be a good idea.
Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 3 above)
Comment (rloterh) [reply app.rb]: well resolved! ✔️

**Pull request merged**